pkgs/applications/terminal-emulators/mlterm/default.nix

```diff
@@ -1,14 +1,82 @@
-{ stdenv, lib, fetchFromGitHub, pkg-config, autoconf, makeDesktopItem, nixosTests
-, libX11, gdk-pixbuf, cairo, libXft, gtk3, vte
-, harfbuzz #substituting glyphs with opentype fonts
-, fribidi, m17n_lib #bidi and encoding
+{ stdenv
+, lib
+, fetchFromGitHub
+, pkg-config
+, autoconf
+, makeDesktopItem
+, nixosTests
+, vte
+# Open Type layout support, (substituting glyphs with opentype fonts)
+, enableOtl ? true
+, harfbuzz # can be replaced with libotf
+, enableBidi ? true
+, fribidi
+, enableM17n ? !stdenv.isDarwin
+, m17n_lib
+, enableSsh2 ? true
 , libssh2 #build-in ssh
-, fcitx5, fcitx5-gtk, ibus, uim #IME
+, enableFcitx ? !stdenv.isDarwin
+, fcitx5, fcitx5-gtk
```

Comment:
Nit: I'd prefer arguments to be grouped by flags and deps rather than mixed like this.

Reply:
Also

```suggestion
, enableFcitx ? !stdenv.isDarwin
, fcitx5
, fcitx5-gtk
```

Reply:
Done.